### ISSUE TITLE
ag-grid 18.0.1 and angular 6 freeze on firefox

### ISSUE THREAD

**Issue by dragouf** [ag-grid/ag-grid#2463]
When I use ng-grid with firefox I got freeze when I try to navigate by clicking on a router navigate link.

some time later firefox tell me that I can kill the running script. When I do it I get this in the console : 

```
Error: Script terminated by timeout at:
__spread@http://localhost:39041/vendor.js:153021:14
debugCheckAndUpdateNode@http://localhost:39041/vendor.js:47780:52
debugCheckRenderNodeFn@http://localhost:39041/vendor.js:47766:13
View_NgbTabset_0/<@ng:///NgbTabsetModule/NgbTabset.ngfactory.js:92:5
debugUpdateRenderer@http://localhost:39041/vendor.js:47758:12
checkAndUpdateView@http://localhost:39041/vendor.js:47134:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execComponentViewsAction@http://localhost:39041/vendor.js:47312:13
checkAndUpdateView@http://localhost:39041/vendor.js:47135:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execComponentViewsAction@http://localhost:39041/vendor.js:47312:13
checkAndUpdateView@http://localhost:39041/vendor.js:47135:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execEmbeddedViewsAction@http://localhost:39041/vendor.js:47333:17
checkAndUpdateView@http://localhost:39041/vendor.js:47130:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execComponentViewsAction@http://localhost:39041/vendor.js:47312:13
checkAndUpdateView@http://localhost:39041/vendor.js:47135:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execEmbeddedViewsAction@http://localhost:39041/vendor.js:47333:17
checkAndUpdateView@http://localhost:39041/vendor.js:47130:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execComponentViewsAction@http://localhost:39041/vendor.js:47312:13
checkAndUpdateView@http://localhost:39041/vendor.js:47135:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execEmbeddedViewsAction@http://localhost:39041/vendor.js:47333:17
checkAndUpdateView@http://localhost:39041/vendor.js:47130:5
callViewAction@http://localhost:39041/vendor.js:47370:21
execComponentViewsAction@http://localhost:39041/vendor.js:47312:13
checkAndUpdateView@http://localhost:39041/vendor.js:47135:5
callWithDebugContext@http://localhost:39041/vendor.js:48022:22
debugCheckAndUpdateView@http://localhost:39041/vendor.js:47700:12
./node_modules/@angular/core/fesm5/core.js/ViewRef_.prototype.detectChanges@http://localhost:39041/vendor.js:45516:13
./node_modules/@angular/core/fesm5/core.js/ApplicationRef.prototype.tick/<@http://localhost:39041/vendor.js:41393:58
./node_modules/@angular/core/fesm5/core.js/ApplicationRef.prototype.tick@http://localhost:39041/vendor.js:41393:13
next/<@http://localhost:39041/vendor.js:41285:99
./node_modules/zone.js/dist/zone.js/</ZoneDelegate.prototype.invoke@http://localhost:39041/polyfills.js:2792:17
onInvoke@http://localhost:39041/vendor.js:40636:24
./node_modules/zone.js/dist/zone.js/</ZoneDelegate.prototype.invoke@http://localhost:39041/polyfills.js:2791:17
./node_modules/zone.js/dist/zone.js/</Zone.prototype.run@http://localhost:39041/polyfills.js:2542:24
./node_modules/@angular/core/fesm5/core.js/NgZone.prototype.run@http://localhost:39041/vendor.js:40550:16
next@http://localhost:39041/vendor.js:41285:69
./node_modules/@angular/core/fesm5/core.js/EventEmitter.prototype.subscribe/schedulerFn<@http://localhost:39041/vendor.js:40367:36
./node_modules/rxjs/_esm5/internal/Subscriber.js/SafeSubscriber.prototype.__tryOrUnsub@http://localhost:39041/vendor.js:130353:13
./node_modules/rxjs/_esm5/internal/Subscriber.js/SafeSubscriber.prototype.next@http://localhost:39041/vendor.js:130291:17
./node_modules/rxjs/_esm5/internal/Subscriber.js/Subscriber.prototype._next@http://localhost:39041/vendor.js:130235:9
./node_modules/rxjs/_esm5/internal/Subscriber.js/Subscriber.prototype.next@http://localhost:39041/vendor.js:130212:13
./node_modules/rxjs/_esm5/internal/Subject.js/Subject.prototype.next@http://localhost:39041/vendor.js:129978:17
./node_modules/@angular/core/fesm5/core.js/EventEmitter.prototype.emit@http://localhost:39041/vendor.js:40359:54
checkStable@http://localhost:39041/vendor.js:40605:13
onLeave@http://localhost:39041/vendor.js:40672:5
onInvoke@http://localhost:39041/vendor.js:40639:17
./node_modules/zone.js/dist/zone.js/</ZoneDelegate.prototype.invoke@http://localhost:39041/polyfills.js:2791:17
./node_modules/zone.js/dist/zone.js/</Zone.prototype.runGuarded@http://localhost:39041/polyfills.js:2555:28
./node_modules/zone.js/dist/zone.js/</Zone.prototype.wrap/<@http://localhost:39041/polyfills.js:2533:24
```

**Comment by LiamDotPro** [comment]
Duplicate of #2457 I raised this issue via enterprise 

AG-1876 | Animation example doesn't work for Angular and React in the website | DONE
-- | -- | --

Looks like they've fixed whatever issue is causing the new version to break on anything but chrome..

**Comment by macqbat** [comment]
i got the same issue on 18.0.1 and 18.1.0 as well. During investigations i found that crash of firefox occure during registration of some beans. When i comment out BodyDropTarget and MoveColumnController freeze dosen't appear anymore, however rows are empty :-(

**Comment by macqbat** [comment]
@vitodata Yes, i can confirm is the truest truth!. After commenting out MutationObserver in resizeObserver FF dosent crash anymore and rows are full as they should. In Chrome this part of code isn't include to run, so that's why Chrome is free of the issue.

**Comment by enniob** [comment]
Is there a fix for this? I'm seeing the same issue when upgrading to the latest version and using Angular 6.

**Comment by seanlandsman** [comment]
I've just done a beta release for this - ag-grid 18.1.2-beta.1

I'd encourage you all to try and see if this resolves the issue 

thanks!

**Comment by enniob** [comment]
@seanlandsman thanks I will download it here and try it.

**Comment by seanlandsman** [comment]
Release 18.1.2 of ag-grid fixes this issue.  Thanks for reporting this.

**Comment by makinggoodsoftware** [comment]
Hi,



Just to let you know that we are working in a solution for this, we are hoping to have it available in our next release, that should be available in the next 2/3 weeks.



I am afraid that for the time being the only solution would be to downgrade to v17 or to not combine ag-grid with ng-bootstrap



Thanks

**Comment by RodSmithAG** [comment]
Hi,

This was released as part of release 18.1.0

You can follow the status of feature requests, bugs and releases using our pipeline: 
https://www.ag-grid.com/ag-grid-pipeline/

And released items in our changelog:
https://www.ag-grid.com/ag-grid-changelog/

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.
